src/eth_provider/provider.rs

```diff
@@ -233,7 +233,7 @@ where
             return Ok(None);
         }
 
-        let filter = into_filter("tx.blockNumber", block_number, 64);
+        let filter = into_filter("tx.blockNumber", block_number, 16);
```

Comment:
Any reason for all these changes? The value of `64` was used in order to pad the hex representation of the value up to `64`, meaning 32 bytes. I don't see any changes in the `into_filter` function so not sure why we need this.